Packs/FreshDesk/Integrations/FreshDesk/FreshDesk.py

```diff
@@ -775,29 +784,43 @@ def test_module():
 
 
 def fetch_incidents():
+    # FreshDesk API supports maximum of 100 tickets per page so if user asked for more, pagination is needed.
+    if MAX_INCIDENTS >= 100:
+        per_page = 100
+    elif MAX_INCIDENTS <= 50:
+        per_page = 50
+    else:
+        per_page = MAX_INCIDENTS
+
     # demisto.getLastRun() will returns an obj with the previous run in it.
     last_run = demisto.getLastRun()
-    # Get the last fetch time, if exists
+    # Get the last fetch time and last id fetched if exist
     last_fetch = last_run.get('last_created_incident_timestamp')
-
+    last_incident_id = last_run.get('last_incident_id', -1)
     # Handle first time fetch, fetch incidents retroactively
     if not last_fetch:
         last_fetch, _ = parse_date_range(FETCH_TIME, to_timestamp=True)
-    updated_since = timestamp_to_datestring(last_fetch, date_format='%Y-%m-%dT%H:%M:%SZ')
-    args = {'updated_since': updated_since, 'order_type': 'asc'}
+    updated_since = timestamp_to_datestring(last_fetch, date_format='%Y-%m-%dT%H:%M:%S')
+    args = {'updated_since': updated_since, 'order_type': 'asc', 'per_page': per_page}
 
-    tickets = search_tickets(args)
+    response = search_tickets(args)  # page 1
+    # handle pagination until user's limit
+    tickets = handle_search_tickets_pagination(args, response, limit=MAX_INCIDENTS)
     # convert the ticket/events to demisto incidents
     incidents = []
     for ticket in tickets:
         incident = ticket_to_incident(ticket)
+        incident_id = ticket.get('id')
         incident_date = date_to_timestamp(incident.get('occurred'), '%Y-%m-%dT%H:%M:%SZ')
-        # Update last run and add incident if the incident is newer than last fetch
-        if incident_date > last_fetch:
+        # Update last run and add incident if the incident is newer than last fetch and was not fetched before
+        if incident_date >= last_fetch and incident_id > last_incident_id:
             last_fetch = incident_date
             incidents.append(incident)
+            last_incident_id = incident_id
+        if len(incidents) == MAX_INCIDENTS:
```

Comment:
to be on the safe side
```suggestion
        if len(incidents) >= MAX_INCIDENTS:
```

Reply:
Done